include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td

```diff
@@ -501,13 +501,17 @@ We call each individual tile "rep".
     InterfaceMethod<"Return total element size per thread.",
                     "unsigned",
                     "getTotalElemsPerThread",
-                     (ins "ArrayRef<int64_t>":$tensorShape,
-                          "Type":$eltTy)>,
+                     (ins "ArrayRef<int64_t>":$shape),
+                     /*defaultImplementation=*/[{
+                         return toLinearEncoding($_self, shape).getTotalElemsPerThread(shape);
+                     }]>,
```

Comment:
In the process of making DistributedEncoding inherit from LinearEncoding, we'll have some funny dependencies like this one, where something is implemented in terms of its children.
This will just be temporary.